src/OrchardCore.Modules/OrchardCore.Queries/Razor/QueryOrchardRazorHelperExtensions.cs

```diff
@@ -25,7 +25,10 @@ public static async Task<IEnumerable> QueryAsync(this IOrchardHelper orchardHelp
             return null;
         }
 
-        var result = await queryManager.ExecuteQueryAsync(query, parameters);
+        var querySource = orchardHelper.HttpContext.RequestServices.GetRequiredKeyedService<IQuerySource>(query.Source);
+
+        var result = await querySource.ExecuteQueryAsync(query, parameters);
```

Comment:
Why not run the query with  `IQueryManager.ExecuteQueryAsync()` that'll resolve the `IQuerySource` already instead?

Also below and in `QueryGlobalMethodProvider`.

Reply:
I used the QueryManager